flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarFunctionCallGen.scala

```diff
@@ -48,10 +48,16 @@ class ScalarFunctionCallGen(
       .getOrElse(throw new CodeGenException("No matching signature found."))
     val resultClass = getResultTypeClass(scalarFunction, matchingSignature)
 
+    // zip for variable signatures
+    var paramToOperands = matchingSignature.zip(operands)
+    var i = paramToOperands.length
+    while (i < operands.length) {
+      paramToOperands = paramToOperands :+ (matchingSignature.head, operands(i))
```

Comment:
Why take head of `matchingSignature`? Isn't the vararg the last one of signatures ? 

And I think we should take the component type of the vararg array type to be the operand class.